src/main/java/com/hotels/beeju/core/BeejuCore.java

```diff
@@ -75,11 +87,27 @@ public BeejuCore(String databaseName, Map<String, String> configuration) {
       // overriding default derby log path to go to tmp
       String derbyLog = File.createTempFile("derby", ".log").getCanonicalPath();
       System.setProperty("derby.stream.error.file", derbyLog);
+
+      //Creating temporary folder
+      init();
     } catch (IOException e) {
       throw new RuntimeException(e);
     }
   }
 
+  /**
+   * Initialise the warehouse path.
+   * <p>
+   * This method can be overridden to provide additional initialisations.
+   * </p>
+   *
+   * @throws Throwable If the initialisation fails.
```

Comment:
Small thing, but it should be lowercase "i" in "If". And it throws `IOException`, not `Throwable`, right?

Reply:
Oh it looks like all the javadocs with `@throws Exception ...` starts with uppercased "i" in "If" so nevermind about that. But I still think the exception should be `IOException` :)

Reply:
Yep good spot!